src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * This TaskList class handles the logic of adding and deleting tasks of Duke
+ * 
+ * @param task Task arraylist to hold the current input tasks
+ * 
+ * @author WangYihe
+ * @author E0424695
+ */
+
+package duke.task;
+
+import duke.ui.Ui;
+import duke.exception.DukeException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+//to handle date and time
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+public class TaskList {
+    public final List<Task> task;
+
+    public TaskList() {
+        task = new ArrayList<>();
+    }
+
+    public int getSize() {
+        return task.size();
+    }
+
+    public void add(Task t) {
+        task.add(t);
+    }
+
+    public Task get(int i) {
+        return task.get(i);
+    }
+
+    public boolean isDateFormat(String date, String pattern) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat(pattern);
+        dateFormat.setLenient(false);
+        if (date.length() != 10) {
+            return false;
+        }
+        try {
+            dateFormat.parse(date.trim());
+        } catch (ParseException e) {
+            //TODO: handle exception
+            return false;
+        }
+        return true;
+    }
+
+    public void add(String[] userInput, Ui ui) throws DukeException {
+        switch (userInput[0]) {
+            case "todo":
+                Todo t = new Todo(userInput[1]);
+                task.add(t);
+                ui.reportTask(t, this);
+                break;
+
+            case "deadline":
+                String[] deadlineArr = userInput[1].split(" /by ", 2);
+                if (deadlineArr.length != 2) {
+                    throw new DukeException("Missing component: due date");
+                }
+                String time = deadlineArr[1];
+                if (isDateFormat(time, "yyyy-mm-dd") || isDateFormat(time, "yyyy-m-dd") || isDateFormat(time, "yyyy-mm-d") || isDateFormat(time, "yyyy-m-d")) {
```

Comment:
Perhaps might want to add more spaces in front to indent this method to fit with the Java coding convention.